src/services/auth.service.ts

```diff
@@ -0,0 +1,60 @@
+import { ICreateUser, ILoginData } from '../types/IUser';
+import { userService, UserService } from './user.service';
+import bcrypt from 'bcrypt';
+import jwt, { Secret } from 'jsonwebtoken';
+
+export class AuthService {
+  constructor(private userService: UserService) {}
+
+  async logIn({ email, password, name }: ILoginData) {
+    const user = await this.userService.getUserByEmail(email);
+
+    if (!user) {
+      throw new Error('user not found');
+    }
+    const match = await bcrypt.compare(password, user.password);
+    if (!match) {
+      throw new Error('wrong password');
+    }
+
+    const accessToken = jwt.sign(
```

Comment:
I think generate tokens should be another service

Reply:
addressed